### PR TITLE
DOC: update documentation on how to build NumPy

### LINKED ISSUE #15151: Docs don't mention that NumPy 1.18 requires cython for build
Install documentation doesn't mention that Cython is now required to build wheels (as implemented in https://github.com/numpy/numpy/commit/d630d96598f1b7ea044d60bea8cd2b87778f072a).

https://numpy.org/devdocs/user/building.html:

> To build development versions of NumPy, you’ll need a recent version of Cython. Released NumPy sources on PyPi include the C files generated from Cython code, so for released versions having Cython installed isn’t needed.

### PR DESCRIPTION
**Pull request by rgommers**
Closes gh-15151

[ci skip]

### CODE REVIEW

File: INSTALL.rst.txt [hunk 1]
```diff
@@ -14,24 +14,18 @@ Prerequisites
...
 Building NumPy requires the following installed software:
 
-1) For Python 3, Python__ 3.5.x or newer.
+1) Python__ 3.6.x or newer.
 
-   On Debian and derivative (Ubuntu): python python-dev
+   Please note that the Python development headers also need to be installed.
+   This is not always the case, e.g. on Debian/Ubuntu one needs to install two
+   packages: `python` and `python-dev`.
+   On Windows and macOS this is normally not an issue.
```
Comment (mattip) [inline INSTALL.rst.txt]: Maybe ".. also need to be installed, e.g., on Debian/Ubuntu one needs to install both `python3` and `python3-dev`."

Comment (rgommers) [reply INSTALL.rst.txt]: thanks, updated to that

Comment (mattip) [reply INSTALL.rst.txt]: I suggested something different. "This" may not be clear to non-native English speakers who may think you mean that the headers themselves do not always need to be installed.

Comment (rgommers) [reply INSTALL.rst.txt]: updated again

**Comment by mattip** [comment]
Thanks @rgommers

File: INSTALL.rst.txt [hunk 2]
```diff
@@ -14,24 +14,17 @@ Prerequisites
...
-
-   Make sure that the Python package distutils is installed before
-   continuing. For example, in Debian GNU/Linux, distutils is included
-   in the python-dev package.
-
-   Python must also be compiled with the zlib module enabled.
-
-2) Cython >= 0.29.2 (for development versions of numpy, not for released
-                     versions)
+2) Cython >= 0.29.13
```
Comment (charris) [inline INSTALL.rst.txt]: Hmm, `0.29.14` is the latest and has a Python 3.8 fix.

Comment (rgommers) [reply INSTALL.rst.txt]: `pyproject.toml` says 0.29.13 in master, do we need to bump it?

Comment (charris) [reply INSTALL.rst.txt]: Yes, we should do that. I'll make a pr.

Comment (charris) [reply INSTALL.rst.txt]: See #15160. I also updated the version here.

**Pull request merged**